lib/inferno/dsl/must_support_metadata_extractor.rb

```diff
@@ -0,0 +1,338 @@
+          {
+            type: 'requiredBinding',
+            values: extract_required_binding_values(pattern_element, metadata)
+          }
+        else
+          # prevent errors in case an IG does something different
+          {
+            type: 'unsupported'
+          }
+        end
+      end
+
+      def required_binding_pattern?(pattern_element)
+        pattern_element.binding&.strength == 'required' && pattern_element.binding&.valueSet
+      end
+
+      def extract_required_binding_values(pattern_element, metadata)
+        value_extractor = ValueExtractor.new(ig_resources, resource, profile_elements)
+
+        value_extractor.codings_from_value_set_binding(pattern_element).presence ||
+          value_extractor.values_from_resource_metadata([metadata[:path]]).presence || []
+      end
+
+      def must_support_type_slice_elements
+        must_support_slice_elements.select do |element|
+          discriminators(sliced_element(element))&.first&.type == 'type'
+        end
+      end
+
+      def type_slices
+        must_support_type_slice_elements.map do |current_element|
+          discriminator = discriminators(sliced_element(current_element)).first
+          type_path = discriminator.path
+          type_path = '' if type_path == '$this'
+          type_element = find_element_by_discriminator_path(current_element, type_path)
+
+          type_code = type_element.type.first.code
+
+          {
+            slice_id: current_element.id,
+            slice_name: current_element.sliceName,
+            path: current_element.path.gsub("#{resource}.", ''),
+            discriminator: {
+              type: 'type',
+              code: type_code.upcase_first
+            }
+          }.tap do |metadata|
+            metadata[:by_requirement_extension_only] = true if by_requirement_extension_only?(current_element)
+          end
+        end
+      end
+
+      def must_support_value_slice_elements
+        must_support_slice_elements.select do |element|
+          discriminators(sliced_element(element))&.first&.type == 'value'
+        end
+      end
+
+      def value_slices
+        must_support_value_slice_elements.map do |current_element|
+          {
+            slice_id: current_element.id,
+            slice_name: current_element.sliceName,
+            path: current_element.path.gsub("#{resource}.", ''),
+            discriminator: {
+              type: 'value'
+            }
+          }.tap do |metadata|
+            metadata[:discriminator][:values] = discriminators(sliced_element(current_element)).map do |discriminator|
+              discriminator_path = discriminator.path
+              discriminator_path = '' if discriminator_path == '$this'
+              fixed_element = find_element_by_discriminator_path(current_element, discriminator_path)
+
+              {
+                path: discriminator_path,
+                value: fixed_element.fixedUri || fixed_element.fixedCode
```

Comment:
The same here. fixedCode and fixUri are the two fixed[x] used by US Core. Other IGs could have additional fixed[x] value.